private/model/api/shape.go

```diff
@@ -201,6 +210,9 @@ func (s *Shape) GoType() string {
 // GoType returns a shape ref's Go type.
 func (ref *ShapeRef) GoType() string {
 	if ref.Shape == nil {
+		if ref.JSONValue {
+			return "interface{}"
```

Comment:
Would this cause a JSONValue type to be rendered as a `interface{}`?

```go
type MyType struct {
    A interface{}
}
```

It looks like it would here. If the shape's type is a `jsonvalue` we can probably treat it as an opaque type and not transverse it at all, and just render as `map[string]interface{}`. This would probably save some additional conditionals/checks that were added.